src/db/tables/onerep_scans.ts

```diff
@@ -13,26 +13,48 @@ import { OnerepScanResultRow, OnerepScanRow } from "knex/types/tables";
 
 const knex = initKnex(knexConfig);
 export interface LatestOnerepScanData {
-  scan: OnerepScanRow | null;
   results: OnerepScanResultRow[];
 }
 
+async function getAllScansForProfile(
+  onerepProfileId: number,
+): Promise<OnerepScanRow[]> {
+  const scans = await knex("onerep_scans")
+    .where("onerep_profile_id", onerepProfileId)
+    .orderBy("created_at", "desc");
+
+  return scans;
+}
+
+async function getScanResults(
+  onerepScanId: number,
+): Promise<OnerepScanResultRow[]> {
+  const scanResults = await knex("onerep_scan_results").where(
+    "onerep_scan_id",
+    onerepScanId,
+  );
+
+  return scanResults;
+}
+
 async function getLatestOnerepScanResults(
   onerepProfileId: number,
 ): Promise<LatestOnerepScanData> {
-  const scan = await knex("onerep_scans")
-    .first()
+  const result = await knex("onerep_scan_results")
+    .select("onerep_scan_id")
+    .from("onerep_scans")
     .where("onerep_profile_id", onerepProfileId)
     .orderBy("created_at", "desc");
 
+  const scanIds = result.map((scan_id) => scan_id["onerep_scan_id"]);
+
   const results =
-    typeof scan === "undefined"
+    typeof scanIds === "undefined"
       ? []
       : await knex("onerep_scan_results")
           .select()
-          .where("onerep_scan_id", scan.onerep_scan_id);
+          .whereIn("onerep_scan_id", scanIds);
   return {
-    scan: scan ?? null,
```

Comment:
There are a couple of places where we rely on the scan returned by this function. We’ll still need a way to get the most recent scan to determine the correct status within the app — for example:

- https://github.com/mozilla/blurts-server/blob/main/src/app/(proper_react)/redesign/(authenticated)/user/dashboard/View.tsx#L85
- https://github.com/mozilla/blurts-server/blob/main/src/app/functions/server/getRelevantGuidedSteps.ts#L170

There are also some stories that need to be updated to conform to the new structure:
- https://github.com/mozilla/blurts-server/blob/main/src/app/(proper_react)/redesign/(authenticated)/user/dashboard/Dashboard.stories.tsx#L118
- https://github.com/mozilla/blurts-server/blob/main/src/app/(proper_react)/redesign/(authenticated)/user/dashboard/fix/data-broker-profiles/view-data-brokers/ViewDataBrokers.stories.tsx#L62

Reply:
Thanks - I couldn't find where that was actually used from a quick look.